### ISSUE TITLE
Optionally augment classifier with "like" fields

### ISSUE THREAD

**Issue by nmittler** [trustin/os-maven-plugin#12]
Now that the os-maven-plugin supports os "like" properites for Linux (https://github.com/trustin/os-maven-plugin#custom-classifiers-for-specific-releases-of-linux), it would be nice to be able to simplify the configuration for apps that us it.  For example, if using `netty-tcnative`, an application will have to do some fancy logic to configure their own custom qualifier to get the right jar. In Maven this is especially ugly as it requires something on the order of a custom ant task.

I'm proposing a new property to the os-maven-plugin to allow the application to automatically augment the `${os.detected.classifier}` with one of a provided set of "likes".  It would work something like this:

```xml
<project>
  <build>
    <extensions>
      <extension>
        <groupId>kr.motd.maven</groupId>
        <artifactId>os-maven-plugin</artifactId>
        <version>1.2.3.Final</version>
        <configuration>
            <!-- List of "likes" that will be appended if the like-property exists. -->
            <!-- First one wins." -->
            <classifierWithLikes>debian, rhel</classifierWithLikes>
        </configuration>
      </extension>
    </extensions>
  </build>
</project>
```

This will generate an `${os.detected.classifier}` of `linux-<arch>-debian` for debian-like systems, `linux-<arch>-redhat` for redhat-like systems, and the default of `<name>-<arch>` for all other systems.

@trustin @normanmaurer WDYT?

**Comment by trustin** [comment]
I like it. Are you going to send a pull request?

**Comment by normanmaurer** [comment]
@nmittler sounds like a very good idea! Hack away :+1:

**Comment by nmittler** [comment]
@normanmaurer @trustin cool ... rolling up my sleeves now :)

**Comment by nmittler** [comment]
@normanmaurer @trustin I created #13 to handle this

**Comment by nmittler** [comment]
closed via #13
